PowerShell/ScubaGear/Modules/Providers/ExportAADProvider.psm1

```diff
@@ -111,21 +111,37 @@ function Export-AADProvider {
         # The RequiredServicePlan variable is used so that PIM Cmdlets are only executed if the tenant has the premium license
         $RequiredServicePlan = $ServicePlans | Where-Object -Property ServicePlanName -eq -Value "AAD_PREMIUM_P2"
 
-        # Get-PrivilegedUser provides a list of privileged users and their role assignments.
         if ($RequiredServicePlan) {
             # If the tenant has the premium license then we also include calls to PIM APIs
-            $PrivilegedUsers = $Tracker.TryCommand("Get-PrivilegedUser", @{"TenantHasPremiumLicense"=$true; "M365Environment"=$M365Environment})
+            $PrivilegedObjects = $Tracker.TryCommand("Get-PrivilegedUser", @{"TenantHasPremiumLicense"=$true; "M365Environment"=$M365Environment})
         }
-        else{
-            $PrivilegedUsers = $Tracker.TryCommand("Get-PrivilegedUser", @{"TenantHasPremiumLicense"=$false; "M365Environment"=$M365Environment})
+        else {
+            $PrivilegedObjects = $Tracker.TryCommand("Get-PrivilegedUser", @{"TenantHasPremiumLicense"=$false; "M365Environment"=$M365Environment})
         }
-        # The Converto-Json call below doesn't need to have the input wrapped in an
-        # array (e.g, "ConvertTo-Json (@PrivilegedUsers)") because $PrivilegedUsers is
-        # a dictionary, not an array, and ConvertTo-Json doesn't mess up dictionaries like it does arrays
-        $PrivilegedUsers = $PrivilegedUsers | ConvertTo-Json
+
+        # # Split the objects into users and service principals
+        $PrivilegedUsers = @{}
+        $PrivilegedServicePrincipals = @{}
+
+        #PrivilegedObjects is an array because of the tracker.trycommand, and so the first index is the hashtable
+        foreach ($key in $PrivilegedObjects[0].Keys) {
```

Comment:
Are we certain that $PrivilegedObjects will always contain at least one entry? Otherwise accessing the 0th index may throw an error if an entry does not exist.

Either way, I also think it would help to see some error handling to check if $PrivilegedObjects[0].Keys is not null prior to entering the loop.